src/bootstrap/config.rs

```diff
@@ -1610,6 +1610,10 @@ impl Config {
         self.submodules.unwrap_or(rust_info.is_managed_git_subrepository())
     }
 
+    pub fn default_codegen_backend(&self) -> Interned<String> {
+        self.rust_codegen_backends.get(0).cloned().unwrap_or(INTERNER.intern_str("llvm"))
+    }
```

Comment:
Could you use this function too in `rustc_cargo_env` to set `CFG_DEFAULT_CODEGEN_BACKEND`? Also if `rust_codegen_backends` is empty, no backend will be built at all, not even LLVM, so the `unwrap_or` should be `unwrap_or(false)` I think.

Reply:
Good idea, done :)